app/serializers/activity_serializer.rb

```diff
@@ -15,7 +15,9 @@ class ActivitySerializer
 
   attribute "prov:wasAttributedTo" do |object|
     url = Rails.env.production? ? "https://api.datacite.org" : "https://api.test.datacite.org"
-    object.username.include?(".") ? url + "/clients/" + object.username : url + "/providers/" + object.username
+    if object.username.present?
```

Comment:
this code doesn't handle object.username.nil? You could do that at the top of the method, e.g. 

```
attribute "prov:wasAttributedTo" do |object|
  return nil if object.username.blank?
```

The rest of the code then doesn't need to change